test/integration/tasks/delete-stale-environments.js

```diff
@@ -0,0 +1,40 @@
+const getCurrentSpace = require('../contentful-client').getCurrentSpace
+const printStepTitle = require('../utils').printStepTitle
+
+const ONE_DAY_IN_MS = 60 * 60 * 24 * 1000
+
+module.exports = async (currentSpace = getCurrentSpace) => {
+  printStepTitle('Removing stale environments')
+
+  const space = await currentSpace()
+  const environments = await space.getEnvironments()
+  const { items } = environments
+
+  // filter for relevant environments
+  const regexps = [/^master$/g, /test/g]
+  const isProtected = name => regexps.some(rgx => new RegExp(rgx).test(name))
+
+  const isStaleEnvironment = timeStamp => {
+    const environmentDate = new Date(timeStamp).getTime()
+    const difference = new Date().getTime() - environmentDate
+    return difference >= ONE_DAY_IN_MS
+  }
+  const deletedEnvironmentIds = []
+
+  items.forEach(async environment => {
```

Comment:
1. I would run all deletions in parallel and wait for all of them to either finish or fail. 
2. If any of deletions fail - we don't care. I think and the race condition between different CI jobs is unavoidable so there will be some failed responses for sure.

Reply:
good point ! Will add that

Reply:
I'd actually run them in series... Reason: rate limits

Reply:
Ya I learned it the hard way. Changed it to run them in a sequential order.